keccak-hash/src/lib.rs

```diff
@@ -50,44 +50,52 @@ pub fn keccak<T: AsRef<[u8]>>(s: T) -> H256 {
 pub unsafe fn keccak_256_unchecked(out: *mut u8, outlen: usize, input: *const u8, inputlen: usize) {
 	// This is safe since `keccak_*` uses an internal buffer and copies the result to the output. This
 	// means that we can reuse the input buffer for both input and output.
-	Keccak::keccak256(slice::from_raw_parts(input, inputlen), slice::from_raw_parts_mut(out, outlen));
+	let input = slice::from_raw_parts(input, inputlen);
+	let output = slice::from_raw_parts_mut(out, outlen);
+	keccak_256(input, output);
 }
 
 pub unsafe fn keccak_512_unchecked(out: *mut u8, outlen: usize, input: *const u8, inputlen: usize) {
 	// This is safe since `keccak_*` uses an internal buffer and copies the result to the output. This
 	// means that we can reuse the input buffer for both input and output.
-	Keccak::keccak512(slice::from_raw_parts(input, inputlen), slice::from_raw_parts_mut(out, outlen));
+	let input = slice::from_raw_parts(input, inputlen);
+	let output = slice::from_raw_parts_mut(out, outlen);
```

Comment:
not sure if this is UB because of potential aliasing

Reply:
no unsafe code is needed here, if you just use the new api

```rust
fn keccak512(data: &mut [u8]) {
	let mut keccak512 = Keccak::v512();
	keccak512.update(&data);
	keccak512.finalize(&mut data);
}
```